extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -283,6 +283,10 @@ void InvokeHTTP::onTrigger(const std::shared_ptr<core::ProcessContext> &context,
   // create a transaction id
   std::string tx_id = generateId();
 
+  std::unique_ptr<utils::ByteInputCallBack> callback = nullptr;
+  std::unique_ptr<utils::HTTPUploadCallback> callbackObj = nullptr;
+
+  // Client declared after the callbacks to make sure the callbacks are still available when the client is destructed
```

Comment:
Might be worth mentioning that the destruction of `callback` and `callbackObj` also matters.

Reply:
Good point, I have added a comment.  This is a good example when a comment is necessary, but it is also a code smell. :)